src/deploy/scope_managementgroup.ts

```diff
@@ -54,16 +54,19 @@ export async function DeployManagementGroupScope(azPath: string, validationOnly:
     core.info("Validating template...")
     var code = await exec(`"${azPath}" deployment mg validate ${azDeployParameters} -o json`, [], validateOptions);
     if (validationOnly && code != 0) {
-        throw new Error("Template validation failed")
+        throw new Error("Template validation failed.")
     } else if (code != 0) {
         core.warning("Template validation failed.")
     }
 
     // execute the deployment
     core.info("Creating deployment...")
-    await exec(`"${azPath}" deployment mg create ${azDeployParameters} -o json`, [], deployOptions);
+    var deploymentCode = await exec(`"${azPath}" deployment mg create ${azDeployParameters} -o json`, [], deployOptions);
```

Comment:
if you want to use return code, then you should set `ignoreReturnCode: true` as done for validateOption